components/financial_assistance/app/domain/financial_assistance/operations/applications/rrv/ifsv/add_rrv_ifsv_determination.rb

```diff
@@ -60,6 +60,8 @@ def update_applicant(response_app_entity, application)
             def update_applicant_evidence(applicant, status, response_applicant_entity)
               response_income_evidence = response_applicant_entity.income_evidence
               income_evidence = applicant.income_evidence
+              # below is a quick fix to avoid duplicate request results, rrv process needs refactoring
+              return if income_evidence.request_results.detect{|evidence| evidence.action == "RRV Response" && evidence.created_at.to_date >= Date.today - 1}.present?
```

Comment:
This is a fine work around while a bigger refactor is considered.  Great comment noting this!  

The only slight issue I see is that the logging/success message could be misleading now.  It will continue to log 'Successfully updated Applicant with evidence' for all cases, whether or not an update occurred.  

I'm not as familiar with the RRV process, so not sure if these logs/messages are used frequently in debugging, but just noting the minor discrepancy being introduced here.

Reply:
I agree on your view on higher level, as I mentioned in the comment RRV process needs to be revisited this just acts a quick fix for now. At this point fti publishes the same application multiple times based on the count of applicants. There is slight possibility that one of the applicant gets update but not other one. In that case a failure would stop the other applicants from getting updated. I am not sure how much we use the logs messages at this point for it to be misleading, as we already doing duplicate publishing from FTI.